commit 7036e64d36b54e75c6428a3c2aa62db0471f1047
Author: Kenny MacDermid <kenny.macdermid@gmail.com>
Date:   Tue Jul 11 22:45:22 2017 -0300

    Add the -mtls-dialect=gnu2 to compile flags.
    
    When this option exists there is no need to add the tls_model attribute.

diff --git a/configure.ac b/configure.ac
index 1551ded8..39f957b1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -701,6 +701,14 @@ if test "x${je_cv_attribute}" = "xyes" ; then
     JE_CXXFLAGS_ADD([-fvisibility=hidden])
   fi
 fi
+dnl Check for tls-dialect support.
+JE_CFLAGS_ADD([-mtls-dialect=gnu2])
+if test "x$je_cv_cflags_added" = "x-mtls-dialect=gnu2" ; then
+  je_cv_tls_dialect_gnu2=yes
+else
+  je_cv_tls_dialect_gnu2=no
+fi
+JE_CXXFLAGS_ADD([-mtls-dialect=gnu2])
 dnl Check for tls_model attribute support (clang 3.0 still lacks support).
 JE_CFLAGS_SAVE()
 JE_CFLAGS_ADD([-Werror])
@@ -711,7 +719,8 @@ JE_COMPILABLE([tls_model attribute], [],
                foo = 0;],
               [je_cv_tls_model])
 JE_CFLAGS_RESTORE()
-if test "x${je_cv_tls_model}" = "xyes" ; then
+if test "x${je_cv_tls_dialect_gnu2}" = "xno" \
+	-a "x${je_cv_tls_model}" = "xyes" ; then
   AC_DEFINE([JEMALLOC_TLS_MODEL],
             [__attribute__((tls_model("initial-exec")))])
 else
